modules/azure/publicaddress.go

```diff
@@ -0,0 +1,113 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// PublicAddressExists indicates whether the specified AzurePublic Address exists.
+// This function would fail the test if there is an error.
+func PublicAddressExists(t testing.TestingT, publicAddressName string, resGroupName string, subscriptionID string) bool {
+	exists, err := PublicAddressExistsE(t, publicAddressName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// PublicAddressExistsE indicates whether the specified AzurePublic Address exists.
+func PublicAddressExistsE(t testing.TestingT, publicAddressName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get the Public Address
+	_, err := GetPublicIPAddressE(t, publicAddressName, resGroupName, subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// GetPublicAddressIP gets the IP of a Public IP Address. This function would fail the test if there is an error.
```

Comment:
NIT: This is super confusing when considering we also offer a function `GetPublicIPAddress`. Can we rename this to something that is descriptive and very distinct from the other function? What do you think of `GetIPOfPublicIPAddressByName`?

Reply:
Changed, good suggestion since it is specific and states the intention!